src/data/botData.ts

```diff
@@ -457,4 +457,212 @@ export function deleteSession(
             reject(error)
         }
     })
+}
+
+// ------------- LOGS -------------
+
+/**
+ * Takes a Log and converts it to raw data acceptable by the database
+ * 
+ * @param log 
+ * @returns RawLog
+ */
+function deconstructLog(log: Log): RawLog {
+    return {
+        message_id: log.messageId,
+        user_id: log.userId,
+        server_id: log.serverId,
+        command: log.command,
+        command_id: log.commandId,
+        date_created: log.dateCreated.toISOString()
+    } as RawLog
+}
+
+/**
+ * Takes a RawLog and converts it to usable data
+ * 
+ * @param rawLog 
+ * @returns Log
+ */
+function buildLog(rawLog: RawLog): Log {
+    return {
+        messageId: rawLog.message_id,
+        userId: rawLog.user_id,
+        serverId: rawLog.server_id,
+        command: rawLog.command,
+        commandId: rawLog.command_id,
+        dateCreated: new Date(rawLog.date_created)
+    } as Log
+}
+
+/**
+ * Returns a Log object for the given server and module
+ * 
+ * @param messageId
+ * @returns Log | null.
+ */
+function getLogSync(
+    messageId: string
+): Log | null {
+    const rawData = db.prepare(`
+    SELECT message_id, user_id, server_id, command, command_id, date_created
+    FROM logs
+    WHERE message_id = ?`).get(messageId) as RawLog | undefined
+
+    if (rawData === undefined) return null
+
+    return buildLog(rawData)
+}
+
+/**
+ * Returns a Log object for the given server and module
+ * 
+ * @param messageId
+ * @returns Log | null.
+ */
+export function getLog(
+    messageId: string
+): Promise<Log | null> {
+    return new Promise<Log | null>((resolve, reject) => {
+        try {
+            resolve(getLogSync(messageId))
+        } catch (error) {
+            reject(error)   
+        }
+    })
+}
+
+/**
+ * Returns All Logs for a given date range
+ * 
+ * @param startDate
+ * @param endDate
+ * @returns Log[] | null
+ */
+function getLogsByDateSync(
+    startDate: Date,
+    endDate: Date
+): Log[] | null {
+    let parameters: string[] = []
+    let values: string[] = []
+    if (startDate !== undefined)
+    {
+        parameters.push(`date_created >= ?`)
+        values.push(startDate.toISOString())
+    }
+    if (endDate !== undefined)
+    {
+        parameters.push(`date_created <= ?`)
+        values.push(endDate.toISOString())
+    }
+    if (parameters.length == 0)
```

Comment:
endDate and startDate will never be undefined here, so why check?
You can change the argument definitions to 
startDate?: Date
endDate?: Date
to fix this.